src/main/java/mate/academy/springbootintro/service/CategoryServiceImpl.java

```diff
@@ -0,0 +1,54 @@
+package mate.academy.springbootintro.service;
+
+import lombok.RequiredArgsConstructor;
+import mate.academy.springbootintro.dto.CategoryDto;
+import mate.academy.springbootintro.dto.CreateCategoryRequestDto;
+import mate.academy.springbootintro.exception.EntityNotFoundException;
+import mate.academy.springbootintro.mapper.CategoryMapper;
+import mate.academy.springbootintro.model.Category;
+import mate.academy.springbootintro.repository.category.CategoryRepository;
+import org.springframework.stereotype.Service;
+import org.springframework.data.domain.Pageable;
+import java.util.List;
+
+@RequiredArgsConstructor
+@Service
+public class CategoryServiceImpl implements CategoryService {
+    private final CategoryRepository categoryRepository;
+    private final CategoryMapper categoryMapper;
+
+    @Override
+    public List<CategoryDto> findAll(Pageable pageable) {
+        return categoryRepository.findAll(pageable).stream()
+                .map(categoryMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public CategoryDto getById(Long id) {
+        Category category = categoryRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException("Can't find category by id: " + id));
+        return categoryMapper.toDto(category);
+    }
+
+    @Override
+    public CategoryDto save(CreateCategoryRequestDto createCategoryRequestDto) {
+        Category entity = categoryMapper.toEntity(createCategoryRequestDto);
+        return categoryMapper.toDto(categoryRepository.save(entity));
+    }
+
+    @Override
+    public CategoryDto update(Long id, CategoryDto categoryDto) {
+        Category existingCategory = categoryRepository.findById(id)

```

Comment:
You're duplicating the call from `getById` method - extract this code to a private method.